docs/recording-tx.md

```diff
@@ -0,0 +1,57 @@
+# RecordingTransaction recipe
+
+A `RecordingTransaction` is an implementation of `Transaction` that logs all transaction operations
+(i.e GET, SET, or DELETE) to a `TxLog` object for later uses such as exporting data.  The code below
+shows how a RecordingTransaction is created by wrapping a Transaction object:
+
+```java
+RecordingTransactionBase rtx = RecordingTransactionBase.wrap(tx);
+```
+
+A predicate function can be passed to wrap method to select which log entries to record.  The code
+below only records log entries whose column family is `meta`:
+
+```java
+RecordingTransactionBase rtx = RecordingTransactionBase.wrap(tx,
+                               le -> le.getColumn().getFamily().toString().equals("meta"));
+```
+
+After creating a RecordingTransaction, users can use it as they would use a Transaction object.
+When they are done operating on the transaction, they can retrieve the TxLog using the following
+code:
+
```

Comment:
Offline you had mentioned that if rtx.get(...) returns null then nothing is added to the txlog.  Is this mentioned in the docs anywhere?  If not might be nice to mention